web/src/app/common/data/token.ts

```diff
@@ -1,23 +1,40 @@
+import {UtilityService} from "../services/utility.service";
+
 export class Token {
     public tokenname: string;
     public currencysymbol: string;
     public fingerprint: string;
     public amount: number;
+    public selected: boolean = false;
 
     constructor(data: any) {
         if (data != null) {
             if (data.tokenname) {
-                this.tokenname = data.tokenname;
+                try {
+                    // Yes, has to be done twice 
+                    // No this is just in the blockchain metadata here just once
+                    this.tokenname = UtilityService.hexToString(data.tokenname);
```

Comment:
I changed this, we need to decode this just once, the metadata (blockchain) are not the same format we get from the backend, its different data. 
Metadata is also encoded in a different (discord was just an example).
I provide you data definitions for the endpoints.

Reply:
Got it